app/src/org/commcare/android/framework/CommCareActivity.java

```diff
@@ -529,22 +529,34 @@ protected void createErrorDialog(String errorMsg, boolean shouldExit) {
     public static void createErrorDialog(final CommCareActivity activity, String errorMsg,
                                          final boolean shouldExit) {
         String title = StringUtils.getStringRobust(activity, org.commcare.dalvik.R.string.error_occured);
+
         AlertDialogFactory factory = new AlertDialogFactory(activity, title, errorMsg);
         factory.setIcon(android.R.drawable.ic_dialog_info);
-        DialogInterface.OnClickListener buttonListener = new DialogInterface.OnClickListener() {
-            @Override
-            public void onClick(DialogInterface dialog, int i) {
-                switch (i) {
-                    case DialogInterface.BUTTON_POSITIVE:

```

Comment:
This switch isn't needed, right? This listener is always assigned to a single button, so we don't need to check the button type